docs/sql-data-sources-parquet.md

```diff
@@ -252,6 +252,51 @@ REFRESH TABLE my_table;
 
 </div>
 
+## Columnar Encryption
+
+
+Since Spark 3.2, columnar encryption is supported for Parquet tables with Apache Parquet 1.12.
+
+Parquet uses the envelope encryption practice, where the file parts are encrypted with “data encryption keys” (DEKs), and the DEKs are encrypted with “master encryption keys” (MEKs). The DEKs are randomly generated by Parquet for each encrypted file/column. The MEKs are generated, stored and managed in a Key Management Service (KMS) of user’s choice. Parquet-test [package](https://repo1.maven.org/maven2/org/apache/parquet/parquet-hadoop/1.12.0/parquet-hadoop-1.12.0-tests.jar) has a mock KMS implementation that allows to run column encryption and decryption without a KMS server:
+
+<div class="codetabs">
+<div data-lang="scala"  markdown="1">
+{% highlight scala %}
+
+sc.hadoopConfiguration.set("parquet.encryption.kms.client.class" ,
+                           "org.apache.parquet.crypto.keytools.mocks.InMemoryKMS")
+
+// Explicit master keys (base64 encoded) - required only for mock InMemoryKMS
+sc.hadoopConfiguration.set("parquet.encryption.key.list" ,
+                   "keyA:AAECAwQFBgcICQoLDA0ODw== ,  keyB:AAECAAECAAECAAECAAECAA==")
+
+// Activate Parquet encryption, driven by Hadoop properties
+sc.hadoopConfiguration.set("parquet.crypto.factory.class" ,
+                   "org.apache.parquet.crypto.keytools.PropertiesDrivenCryptoFactory")
+
+// Write encrypted dataframe files. 
+// Column "square" will be encrypted with master key "keyA".
+// Parquet file footers will be encrypted with master key "keyB"
+squaresDF.write.
+   option("parquet.encryption.column.keys" , "keyA:square").
+   option("parquet.encryption.footer.key" , "keyB").
+parquet("/path/to/table.parquet.encrypted")
+
+// Read encrypted dataframe files
+val df2 = spark.read.parquet("/path/to/table.parquet.encrypted")
```

Comment:
Don't we need any options for reading encrypted file?

Reply:
Nope. The options are passed in writing only; and stored in Parquet metadata, so the readers don't need them.